src/Common/src/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -25,6 +25,7 @@ public struct PInvokeILEmitter
         private readonly Marshaller[] _marshallers;
         private readonly PInvokeILEmitterConfiguration _pInvokeILEmitterConfiguration;
         private readonly PInvokeMetadata _importMetadata;
```

Comment:
Consider breaking this class into two sub-classes and move _importMetadata to the pinvoke sub-class. We do similar things in desktop/CoreCLR as well.